lib/Doctrine/Common/Annotations/CachedReader.php

```diff
@@ -51,16 +53,24 @@ final class CachedReader implements Reader
     private $loadedAnnotations = [];
 
     /**
-     * Constructor.
-     *
-     * @param Reader $reader
-     * @param Cache  $cache
-     * @param bool   $debug
+     * @param Cache|CacheInterface $cache
+     * @param bool                 $debug
      */
-    public function __construct(Reader $reader, Cache $cache, $debug = false)
+    public function __construct(Reader $reader, $cache, $debug = false)
     {
+        if ($cache instanceof Cache) {
+            @trigger_error(sprintf(
+                'Passing a %s instance to %s is deprecated since doctrine/annotations 1.9 and will not be possible anymore in 2.0. Please pass a %s instance',
+                Cache::class,
+                __METHOD__,
+                CacheInterface::class
+            ), E_USER_DEPRECATED);
+            $cache = new SimpleCacheAdapter($cache);
+        }
         $this->delegate = $reader;
-        $this->cache = $cache;
+        $this->cache = (static function (CacheInterface $cache): CacheInterface {
+            return $cache;
+        })($cache);
```

Comment:
A weird way to not have to generate the error message in case of unsupported type, tell me if it's too weird

Reply:
Maybe using an assertion would be simpler, or just an if? Removes a function call which is good in this context